HeterogeneousCore/AlpakaInterface/interface/prefixScan.h

```diff
@@ -0,0 +1,235 @@
+        if (size <= warpSize)
+          return;
+        if (blockThreadIdx < warpSize) {
+          warpPrefixScan(acc, laneId, ws, blockThreadIdx);
+        }
+        alpaka::syncBlockThreads(acc);
+        for (auto i = first + warpSize; i < size; i += blockDimension) {
+          int32_t warpId = i / warpSize;
+          co[i] += ws[warpId - 1];
+        }
+        alpaka::syncBlockThreads(acc);
+      } else {
+        co[0] = ci[0];
+        for (int32_t i = 1; i < size; ++i)
+          co[i] = ci[i] + co[i - 1];
+      }
+    }
+
+    template <typename TAcc, typename T>
+    ALPAKA_FN_HOST_ACC ALPAKA_FN_INLINE void blockPrefixScan(const TAcc& acc,
+                                                             T* __restrict__ c,
+                                                             int32_t size,
+                                                             T* __restrict__ ws = nullptr) {
+      if constexpr (!requires_single_thread_per_block_v<TAcc>) {
+        const auto warpSize = alpaka::warp::getSize(acc);
+        int32_t const blockDimension(alpaka::getWorkDiv<alpaka::Block, alpaka::Threads>(acc)[0u]);
+        int32_t const blockThreadIdx(alpaka::getIdx<alpaka::Block, alpaka::Threads>(acc)[0u]);
+        ALPAKA_ASSERT_OFFLOAD(ws);
+        ALPAKA_ASSERT_OFFLOAD(size <= warpSize * warpSize);
+        ALPAKA_ASSERT_OFFLOAD(0 == blockDimension % warpSize);
+        auto first = blockThreadIdx;
+        auto laneId = blockThreadIdx & (warpSize - 1);
+
+        for (auto i = first; i < size; i += blockDimension) {
+          warpPrefixScan(acc, laneId, c, i);
+          auto warpId = i / warpSize;
+          ALPAKA_ASSERT_OFFLOAD(warpId < warpSize);
+          if ((warpSize - 1) == laneId)
+            ws[warpId] = c[i];
+        }
+        alpaka::syncBlockThreads(acc);
+        if (size <= warpSize)
+          return;
+        if (blockThreadIdx < warpSize) {
+          warpPrefixScan(acc, laneId, ws, blockThreadIdx);
+        }
+        alpaka::syncBlockThreads(acc);
+        for (auto i = first + warpSize; i < size; i += blockDimension) {
+          auto warpId = i / warpSize;
+          c[i] += ws[warpId - 1];
+        }
+        alpaka::syncBlockThreads(acc);
+      } else {
+        for (int32_t i = 1; i < size; ++i)
+          c[i] += c[i - 1];
+      }
+    }
+
+    // in principle not limited....
+    template <typename T>
+    struct multiBlockPrefixScan {
+      template <typename TAcc>
+      ALPAKA_FN_ACC void operator()(const TAcc& acc,
+                                    T const* ci,
+                                    T* co,
+                                    uint32_t size,
+                                    int32_t numBlocks,
+                                    int32_t* pc,
+                                    std::size_t warpSize) const {
+        // Get shared variable. TODO: The workspace is in fact not needed at all for single thread multi element accelerators TODO
```

Comment:
I think this is properly implements, and so `TODO` can be dropped from the comment ?
```suggestion
        // Get shared variable. The workspace is needed only for multi-threaded accelerators.
```